python/cudf/cudf/core/frame.py

```diff
@@ -3658,6 +3705,125 @@ def _colwise_binop(
 
         return output
 
+    # For more detail on this function and how it should work, see
+    # https://numpy.org/doc/stable/reference/ufuncs.html
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        # We don't currently support reduction, accumulation, etc. We also
+        # don't support any special kwargs or higher arity ufuncs than binary.
+        if method != "__call__" or kwargs or ufunc.nin > 2:
+            return NotImplemented
+
+        # Binary operations
+        binary_operations = {
```

Comment:
Can we move this `dict` construction to outside this method so that we don't reinitialize this every time function is called?

Reply:
Sure. We should aim to be consistent about these kinds of variables in the future; personally I value the readability more than the small speedups (defining this should take < 1 us), but for now I'll make the change and we can come back to this question at another time.

Reply:
> We should aim to be consistent about these kinds of variables in the future;

AFAIK this has been a pattern/trend since the beginning - That we make these kinds of variables that don't change on function execution global. But it is highly possible that some parts of the code-base are doing otherwise, we should probably just make them global as and when we run into modifying those parts of code.

Reply:
Sounds good! Yes, consistency is the most important thing IMO.